utilities/src/main/java/ru/study21/jcsv/xxl/io/CSVOffsetExtractor.java

```diff
@@ -0,0 +1,48 @@
+package ru.study21.jcsv.xxl.io;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.FileChannel;
+
+// writes an array of long-s; i-th long is i-th row END (!)
+// rows are detected with '\n'
+// for CRLF we will copy '\r', then write '\n', so all good
+// (validity of CSV file is not checked)
+public class CSVOffsetExtractor {
+
+    // NOTE: input file MUST have line break at the end!
+    public static void extractOffsets(
+            FileChannel inputChannel,
+            FileChannel outputChannel,
+            int readBufferSize,
+            int writeBufferSize
+    ) throws IOException {
+        ByteBuffer readBuf = ByteBuffer.allocate(readBufferSize);
+        ByteBuffer writeBuf = ByteBuffer.allocate(writeBufferSize);
+        long pos = 0;
+
+        while (inputChannel.read(readBuf) > 0) {
+            readBuf.flip();
+            while (readBuf.hasRemaining()) {
+                byte c = readBuf.get();
+                if (c == '\n') {
```

Comment:
Naive assumption that '\n' means record separator.
'\n' can be part of field

https://datatracker.ietf.org/doc/html/rfc4180

Reply:
```
  field = (escaped / non-escaped)

   escaped = DQUOTE *(TEXTDATA / COMMA / CR / LF / 2DQUOTE) DQUOTE
```

Reply:
Yeah, this is a problem I just didn't have time for yet

Also, in the current version writing offsets requires an extra pass. We can actually output offsets while reading the file with CSVReader (which very conveniently has the info about escaped symbols)